tabular/src/autogluon/tabular/models/abstract/model_trial.py

```diff
@@ -62,13 +70,21 @@ def fit_and_save_model(model, params, fit_args, predict_proba_args, y_val, time_
 
     model.params.update(params)
     time_fit_start = time.time()
-    model.fit(**fit_args, time_limit=time_left, reporter=reporter)
-    time_fit_end = time.time()
-    y_pred_proba = model.predict_proba(**predict_proba_args)
-    time_pred_end = time.time()
-    model.val_score = model.score_with_y_pred_proba(y=y_val, y_pred_proba=y_pred_proba)
-    model.fit_time = time_fit_end - time_fit_start
-    model.predict_time = time_pred_end - time_fit_end
+    incremental = epochs > 1
+    for i in range(epochs):
+        model.fit(**fit_args, time_limit=time_left, reporter=reporter, warm_start=incremental)
+        time_fit_end = time.time()
+        y_pred_proba = model.predict_proba(**predict_proba_args)
+        time_pred_end = time.time()
+        val_score = model.score_with_y_pred_proba(y=y_val, y_pred_proba=y_pred_proba)
+        model.val_score = val_score
+        model.fit_time = time_fit_end - time_fit_start
+        model.predict_time = time_pred_end - time_fit_end
+        if report_probs:
+            #use model.path as a unique identifier. maybe change this later?
+            reporter(epoch=i+1, validation_performance=val_score, y_probs=y_pred_proba, y_val=y_val, model_path=model.path)
```

Comment:
Have you thought about storing the prediction vectors here, instead of sending them back to the scheduler?

Reply:
I mean, you store the model here, why not the prediction vector as well?

Reply:
Will rework the validation probability passing